services/libs/aster-std/src/process/process/process_group.rs

```diff
@@ -69,6 +69,11 @@ impl ProcessGroup {
         self.pgid
     }
 
+    /// Returns all processes in the group.
+    pub fn get_all_processes(&self) -> Vec<Arc<Process>> {
+        self.inner.lock().processes.values().cloned().collect()
```

Comment:
When the operation on each process is light-weight enough (like setting a nice value), dumping all processes into a `Vec` seems too costly. I think a more desirable approach is adding an `iter` method.

```rust
pub struct Iter<'a> {
    inner: MutexGuard<'a, ProcessGroupInner>
}

/// Returns an iterator for the processes within the group.
pub fn iter(&self) -> Iter<'a> {
    ...
}
```


This way the user can enjoy the flexibility of choosing between two approaches:
1. Holding the internal lock while iterating the member processes. No copying or creating `Vec`.
2. Calling the `collect` method on the iterator to return a `Vec`.

I recommend adding the `iter` method to process table, process group, and process.

Reply:
In the beginning, I just followed the method of `get_all_processes` in process_table, the `iter` method is  better.

Done with adding the `iter()` method.